iothub/device/src/DirectMethod/DirectMethodRequest.cs

```diff
@@ -131,9 +140,9 @@ public bool TryGetPayload<T>(out T payload)
         /// cancellationToken);
         /// </code>
         /// </example>
-        public byte[] GetPayloadAsBytes()
+        public byte[] GetPayload()
         {
-            return _payload == null || _payload.Length == 0 ? null : (byte[])_payload.Clone();
```

Comment:
Is there a reason this didn't work?
This class has the payload always available as bytes (only constructor present is `protected internal DirectMethodRequest(string methodName, byte[] payload = default)`). So, this should directly work.

Reply:
Yup, this should work. Changing it back.